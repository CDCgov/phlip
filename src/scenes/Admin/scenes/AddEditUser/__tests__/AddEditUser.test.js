import React from 'react'
import { AddEditUser } from '../index'
import { shallow } from 'enzyme'

const props = {
  users: [],
  avatar: '',
  currentUser: { id: 1, firstName: 'test', lastName: 'user', avatar: '' },
  actions: {
    loadAddEditAvatar: jest.fn(),
    addUserRequest: jest.fn(),
    updateUserRequest: jest.fn(),
    updateCurrentUser: jest.fn()
  },
  formActions: {},
  location: {
    pathname: '/admin/new/user'
  },
  match: {
    params: {}
  },
  history: {},
  onCloseModal: jest.fn(),
  formError: '',
  isDoneSubmitting: false,
  onSubmitError: jest.fn(),
  selectedUser: {}
}

describe('User Management - AddEditUser scene', () => {
  test('should render correctly', () => {
    expect(shallow(<AddEditUser {...props} />)).toMatchSnapshot()
  })
})
