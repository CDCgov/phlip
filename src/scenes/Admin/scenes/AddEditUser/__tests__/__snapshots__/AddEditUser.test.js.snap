// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Management - AddEditUser scene should render correctly 1`] = `
<WithStyles(ModalForm)
  actions={
    Array [
      Object {
        "onClick": [Function],
        "otherProps": Object {
          "aria-label": "Cancel and close form",
        },
        "type": "button",
        "value": "Cancel",
      },
      Object {
        "disabled": undefined,
        "otherProps": Object {
          "aria-label": "Save form",
        },
        "type": "submit",
        "value": <React.Fragment>
          Save
        </React.Fragment>,
      },
    ]
  }
  asyncBlurFields={
    Array [
      "email",
    ]
  }
  asyncValidate={[Function]}
  form="addEditUser"
  handleSubmit={[Function]}
  height="400px"
  initialValues={Object {}}
  onClose={[MockFunction]}
  open={true}
  title="Edit User"
  width="600px"
>
  <FlexGrid
    align="stretch"
    circular={false}
    container={true}
    flex={false}
    justify="stretch"
    padding="30px 15px 0"
    raised={false}
    style={
      Object {
        "minHeight": 275,
        "minWidth": 500,
      }
    }
    type="column"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={false}
      justify="space-between"
      padding="0 0 30px"
      raised={false}
      style={Object {}}
      type="row"
    >
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={false}
        justify="stretch"
        padding="0 30px 0 0"
        raised={false}
        style={Object {}}
        type="column"
      >
        <ImageFileReader
          base64={true}
          fileTypes={
            Array [
              "jpg",
              "png",
            ]
          }
          handleFiles={[Function]}
          maxSize={500000}
        >
          <IconButton
            color="#757575"
            disabled={false}
            iconSize={50}
            id="add-user-photo"
            tooltipText="Add a photo"
          >
            add_a_photo
          </IconButton>
        </ImageFileReader>
      </FlexGrid>
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={true}
        justify="stretch"
        padding="0 20px 0 0"
        raised={false}
        style={Object {}}
        type="column"
      >
        <Field
          component={[Function]}
          fullWidth={true}
          label="First Name"
          name="firstName"
          placeholder="Enter first name"
          required={true}
          shrinkLabel={true}
          validate={[Function]}
        />
      </FlexGrid>
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={true}
        justify="stretch"
        padding={0}
        raised={false}
        style={Object {}}
        type="column"
      >
        <Field
          component={[Function]}
          fullWidth={true}
          label="Last Name"
          name="lastName"
          placeholder="Enter last name"
          required={true}
          shrinkLabel={true}
          validate={[Function]}
        />
      </FlexGrid>
    </FlexGrid>
    <FlexGrid
      align="stretch"
      circular={false}
      container={false}
      flex={false}
      justify="stretch"
      padding="0 0 30px"
      raised={false}
      style={Object {}}
      type="column"
    >
      <Field
        component={[Function]}
        disabled={false}
        fullWidth={true}
        label="Email"
        name="email"
        placeholder="Enter email"
        required={true}
        shrinkLabel={true}
        validate={[Function]}
      />
    </FlexGrid>
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={true}
      justify="stretch"
      padding={0}
      raised={false}
      style={Object {}}
      type="row"
    >
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={true}
        justify="stretch"
        padding={0}
        raised={false}
        style={Object {}}
        type="column"
      >
        <Field
          component={[Function]}
          defaultValue=""
          disabled={false}
          id="role"
          label="Role"
          name="role"
          options={
            Array [
              Object {
                "label": "Admin",
                "value": "Admin",
              },
              Object {
                "label": "Coordinator",
                "value": "Coordinator",
              },
              Object {
                "label": "Coder",
                "value": "Coder",
              },
            ]
          }
          style={
            Object {
              "display": "flex",
            }
          }
        />
      </FlexGrid>
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={true}
        justify="stretch"
        padding={0}
        raised={false}
        style={Object {}}
        type="column"
      >
        <Field
          component={[Function]}
          disabled={false}
          label="Active"
          name="isActive"
          style={
            Object {
              "display": "10px",
            }
          }
        />
      </FlexGrid>
    </FlexGrid>
  </FlexGrid>
  <Route
    component={[Function]}
    path="/user/profile/avatar"
  />
</WithStyles(ModalForm)>
`;
